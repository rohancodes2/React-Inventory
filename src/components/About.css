.search-container {
    padding: 6px;
    margin-top: 8px;
    margin-right: 200px;
    background: rgb(255, 255, 255);
    font-size: 20px;
  
    border: none;
    cursor: pointer;
  }
  #inn {
    height: 30px;
    width: 90%;
  }

.ggg{
    background-color: rgb(255, 255, 255);
  }
  .gggxx{
    background-color: rgb(255, 255, 255);
  }
  
  
  #asd {
    font-family: Georgia, "Times New Roman", Times, serif;
  }
  #customise{
    margin-left: 600px;
  }

  @media screen and (max-width: 800px) {
    #customise {
     margin-left: 0px;
     font-size: 15px ;
    }
    .gggxx{
      width: 500px;
    }
   
  }

