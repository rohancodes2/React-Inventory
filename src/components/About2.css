:root{
    --timeout:3500ms;
}
.hhhhh{
    background-color: rgb(233, 214, 175);
    color:rgb(0, 0, 0);
    
}



.balloon-enter{
opacity: 0;
}
.balloon-enter-active{
opacity: 1;
transition: opacity var(--timeout);
}
.balloon-exit{
opacity: 1;
}
.balloon-exit-active{
opacity:0;
transition: opacity var(--timeout)
}