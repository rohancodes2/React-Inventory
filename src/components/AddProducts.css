:root{
    --timeout:3500ms;
}
.hhhhh{
    background-color: rgb(255, 205, 98);
    color:rgb(255, 111, 0);
    width:1000px;
    height:50px
}



.balloon-enter{
opacity: 0;
}
.balloon-enter-active{
opacity: 1;
transition: opacity var(--timeout);
}
.balloon-exit{
opacity: 1;
}
.balloon-exit-active{
opacity:0;
transition: opacity var(--timeout)
}