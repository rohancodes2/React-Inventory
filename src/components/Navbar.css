a.active {
    background-color:#F7882F;
    color: white;
  }
  .nnn{
      
      background-color:#ffbc2c ;}
    /* #F7882F #ffbc2c */