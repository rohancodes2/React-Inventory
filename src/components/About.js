import React, { useEffect, useState, lazy, Suspense } from "react";
import {
  useHistory,
  Redirect,
  BrowserRouter as Router,
} from "react-router-dom";
import "./About.css";
import { connect } from "react-redux";
import { CSSTransition } from "react-transition-group";
import "./About2.css";
import { SubItems1 } from "../data/NavbarItems";
const ProductList = lazy(() => import("./ProductList"));
export function About(props) {
  let history = useHistory();
  function handler() {
    history.push("/products/view");
  }

  const [products, setProducts] = useState(props.products);
  const [options, setOptions] = useState(props.products);
  const [productType, setPT] = useState([]);
  const [search, setSearch] = useState("");
  const [touch, touched] = useState(false);
  const [customised,customise]= useState([]);
  const [customised2,customise2]= useState([]);
  const [customised3,customise3]= useState([]);
  const [filter3,setFilter3]=useState(false)
  const [filter,setFilter]=useState(false)
  const [filter2,setFilter2]=useState(false)
  const toggle = () => {
    touched(!touch);
  };
  
  const mainFilter2=(m)=>{
    //  if(customised.length>0){
    //    doubleFilter()
    //  }
    console.log("in mainfilter2 products",products)
     console.log(customised2)
     console.log('mainfilter2 called')
    let prods=[]
    
    customised2.map((item)=>{
      if(m){
     prods= prods.concat(
        products.filter((product) => {
        return (
          (product.productDescription
            .indexOf(item) !== -1 )
        );
      }
      ))}else{
        prods= prods.concat(
          products.filter((product) => {
          return (
            (product.productDescription
              .indexOf(item) === -1 )
          );
        }
        ))
      }})
      console.log("prods in mainfilter2",prods)
      setProducts(prods)
  }

  const mainFilter=()=>{
    //  if(customised.length>0){
    //    doubleFilter()
    //  }
     console.log(customised)
     console.log('mainfilter called')
    let prods=[]
    
    customised.map((item)=>{
      
     prods= prods.concat(
        props.products.filter((product) => {
        return (
          (product.manufacturer
            .indexOf(item) !== -1 ||product.productDescription
            .indexOf(item) !== -1 )
        );
      }
      ))})
      console.log("prods in mainfilter1",prods)
      setProducts(prods)
  }
  const mainFilter3=()=>{
    //  if(customised.length>0){
    //    doubleFilter()
    //  }

     console.log('mainfilter3 called')
    let prods=[]
    
    customised3.map((item)=>{
      
     prods= prods.concat(
        products.filter((product) => {
        return (
          (product.price>(parseInt(item)-20000)&&product.price<(parseInt(item)))
            
        );
      }
      ))})
      console.log("prods in mainfilter3",prods)
      setProducts(prods)
  }
  const handleFilter = (e) => {
    console.log(e.target.checked)
    console.log(e.target.value);
   (e.target.checked)?
   customised.push(e.target.value)
  :customise(customised.filter((i)=>{ if(i!==e.target.value){
     return i;
   }}));
  e.target.checked?setFilter(false):setFilter(true)
 mainFilter()
}
const handleFilter2 = (e) => {
  console.log(e.target.checked)
  console.log(e.target.value);
 (e.target.checked)?
 customised2.push(e.target.value)
:customise2(customised2.filter((i)=>{ if(i!==e.target.value){
   return i;
 }}));
 if(e.target.checked){setFilter(false)
  mainFilter2(true)}
 else{ mainFilter2(false); setFilter(true)}
console.log("in handler filter 2",products)

}

const handleFilter3 = (e) => {
  console.log(e.target.checked)
  console.log(e.target.value);
 (e.target.checked)?
 customised3.push(e.target.value)
:customise3(customised3.filter((i)=>{ if(i!==e.target.value){
   return i;
 }}));
e.target.checked?setFilter3(false):setFilter3(true)
console.log("in handler filter 3",products)
mainFilter3()
}

    // let filteredProducts = products.filter((product) => {
    //   return (
    //     (product.manufacturer
    //       .toLowerCase()
    //       .indexOf(e) !== -1 )
    //   );
    // });
    // setProducts(filteredProducts);

  
  const handleClick = () => {
    let filteredProducts = products.filter((product) => {
      return (
        product.productName
          .toLowerCase()
          .indexOf(search.toLowerCase().trim()) !== -1
      );
    });
    setProducts(filteredProducts);
  };

  const onChange = (event) => {
    setSearch(event.target.value);
    console.log("Change recorded");
    console.log(event.target.value);
    handleClick();
   
  };

  const blurit =()=>{
  console.log(touched);
  }
  useEffect(() => {
    if (search === ""&&(customised.length===0&&customised2.length===0&&customised3.length===0)) {
      console.log("in here");
      setProducts(props.products);
      setOptions(props.options);
      setPT(props.productTypes);
     // console.log(customised)
      // setOptions(Array.from(new Set(props.products.map((item)=>{return item.name}))))
     
    }

    if(filter){
      setFilter(false)
      mainFilter()}
      if(filter2){  
        setFilter2(false)
        mainFilter2(true)}
        if(filter3){
          setFilter3(false)
          mainFilter3()}
console.log(customised3)
  });

  return (
    <div class="ggg">
      <div class="search-container ggg  ">
        <div>
          {" "}
          <input
            class="input-text"
            id="inn"
            onChange={onChange}
            type="text"
            placeholder="Search any item..."
          ></input>
          <button
            class="btn btn-dark text-white"
         
            onClick={handleClick}
            
          >
            Search
          </button>
        </div>
      </div>
      <div class="mx-auto" style={{ width: 50 }}>
        <button onClick={toggle} className='btn btn-primary'>Customise your product</button>
      </div>
      <img src='../AboutPageScreenshot.png'>f</img>
      <CSSTransition
        in={touch}
        timeout={500}
        classNames="balloon"
        unmountOnExit
      >
        <div
          className="mx-auto hhhhh p2 row justify-content-center'"
          style={{ width: 700, height: 150,color:'black' }}
        >
          <ul>
            Brands
            <li>
              {options.map((item) => {
                return (
                  <div>
                    <input type="checkbox" value={item}onClick={handleFilter} />
                    <label>{item}</label>
                  </div>
                );
              })}
            </li>
          </ul>
          <div>
            {" "}
            <ul>
              ProductType
              <li>
                {productType.map((item) => {
                  return (
                    <div>
                      <input type="checkbox" value={item} onClick={handleFilter2} />
                      <label>{item}</label>
                    </div>
                  );
                })}
              </li>
            </ul>
          </div>
          <div>
            <ul>
             Price Range
              <li>
                <input type="checkbox" value={20000} onClick={handleFilter3}/>
                <label>0-20000</label>
              </li>
              <li>
                <input type="checkbox" value={40000} onClick={handleFilter3}/>
                <label>20000-40000</label>
              </li>
              <li>
                <input type="checkbox" value={60000} onClick={handleFilter3} />
                <label>40000-60000</label>
              </li>
              <li>
                <input type="checkbox" value={80000} onClick={handleFilter3} />
                <label>60000-80000</label>
              </li>
            </ul>
          </div>
        </div>
      </CSSTransition>
      <Suspense fallback={<div>Loading</div>}>
        <ProductList list={products}></ProductList>
      </Suspense>
    </div>
  );
}
const mapStateToProps = (state) => {
  return {
    products: state.products,
    options: state.brands,
    productTypes: state.productTypes,
  };
};

export default connect(mapStateToProps)(About);
