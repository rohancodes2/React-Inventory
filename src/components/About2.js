import React, { useEffect, useState, lazy, Suspense } from "react";
import {
  useHistory,
  Redirect,
  BrowserRouter as Router,
} from "react-router-dom";
import "./About.css";
import { connect } from "react-redux";
import { CSSTransition } from "react-transition-group";
import "./About2.css";
import { SubItems1 } from "../data/NavbarItems";
const ProductList = lazy(() => import("./ProductList"));
function About2(props) {
  let history = useHistory();
  function handler() {
    history.push("/products/view");
  }

  const [products, setProducts] = useState(props.products);
  const [options, setOptions] = useState(props.products);
  const [productType, setPT] = useState([]);
  const [search, setSearch] = useState("");
  const [touch, touched] = useState(false);
  const [customised, customise] = useState([]);
  const [customised2, customise2] = useState([]);
  const [customised3, customise3] = useState([]);
  const [filter3, setFilter3] = useState(false);
  const [filter, setFilter] = useState(false);
  const [filter2, setFilter2] = useState(false);
  const [flag,setFlag]=useState(false)
  const toggle = () => {
    touched(!touch);
  };

  const handleClick = () => {
    let filteredProducts = products.filter((product) => {
      return (
        product.productName
          .toLowerCase()
          .indexOf(search.toLowerCase().trim()) !== -1
      );
    });
    setProducts(filteredProducts);
  };

  const onChange = (event) => {
    setSearch(event.target.value);
    console.log("Change recorded");
    console.log(event.target.value);
    handleClick();
  };
  const mainFilter=()=>{
    console.log("here in mainfil")
    console.log(customised2)
   let prods=[]
   if(filter){
    customised.map((item)=>{
       prods= prods.concat(props.products.filter((i)=>{
             return( i.manufacturer.indexOf(item)!==-1)
         })) 
       setProducts(prods)
 })}
 if(filter2){
   console.log(filter2)
   let prods2=[]
   if(prods.length===0){prods=props.products}
 customised2.map((item)=>{
  prods2= [...prods2 ,...prods.filter((i)=>{
        return( i.productDescription.indexOf(item)!==-1)
    }) ]
    
    console.log('prods',prods2)
  setProducts(prods2)
})}
if(filter3){
  let prods2=[]
  if(prods.length===0){prods=props.products}
customised3.map((item)=>{
  prods2= [...prods2,...prods.filter((product)=>{
        return(product.price>(parseInt(item)-20000)&&product.price<(parseInt(item)))
    }) ]
  setProducts(prods2)
})

   }

  }


  const handleFilter = (e) => {
    
    switch (e.target.id) {
      case "brands":
        e.target.checked
          ? customise([...customised,e.target.value])
          : customise(
              customised.filter((i) => {
                if (i !== e.target.value) {
                  return i;
                }
              })
            );
          
        break;
      case "types":
        e.target.checked
          ? customise2([...customised2,e.target.value])
          : customise2(
              customised2.filter((i) => {
                if (i !== e.target.value) {
                  return i;
                }
              })
            );
    
        break;
      case "price":
        e.target.checked
          ? customise3([...customised3,e.target.value])
          : customise3(
              customised3.filter((i) => {
                if (i !== e.target.value) {
                  return i;
                }
              })
            );
        break;


    }
   
  //mainFilter()
  e.target.checked?setFlag(true):setFlag(false)
  };

  const blurit = () => {
    console.log(touched);
  };
  useEffect(() => {
    if (
      search === "" &&
      customised.length === 0 &&
      customised2.length === 0 &&
      customised3.length === 0
    ) {
      console.log("in here");
      setProducts(props.products);
      setOptions(props.options);
      setPT(props.productTypes);
     
      // setOptions(Array.from(new Set(props.products.map((item)=>{return item.name}))))
    }
    customised.length>0?setFilter(true):setFilter(false)
    customised2.length>0?setFilter2(true):setFilter2(false)
    customised3.length>0?setFilter3(true):setFilter3(false)
  });

  return (
    <div class="ggg">
      <div class="search-container gggxx  ">
        <div>
          {" "}
          <input
            class="input-text"
            id="inn"
            onChange={onChange}
            type="text"
            placeholder="Search any item..."
          ></input>
          <button class="btn btn-dark text-white" onClick={handleClick}>
            Search
          </button>
        </div>
      </div>
      <div id='customise' style={{ width: 50 }}>
        <button onClick={toggle} className="btn btn-primary">
          Customise your product
        </button>
      </div>

      <CSSTransition
        in={touch}
        timeout={500}
        classNames="balloon"
        unmountOnExit
      >
        <div
          className="mx-auto hhhhh p2 row justify-content-center'"
          style={{ width: 500, height: 200, color: "black" }}
        >
          <ul>
            Brands
            <li>
              {options.map((item) => {
                return (
                  <div>
                    <input
                      type="checkbox"
                      id={"brands"}
                      value={item}
                      onClick={handleFilter}
                    />
                    <label>{item}</label>
                  </div>
                );
              })}
            </li>
          </ul>
          <div>
            {" "}
            <ul>
              ProductType
              <li>
                {productType.map((item) => {
                  return (
                    <div>
                      <input
                        type="checkbox"
                        id={"types"}
                        value={item}
                        onClick={handleFilter}
                      />
                      <label>{item}</label>
                    </div>
                  );
                })}
              </li>
            </ul>
          </div>
          <div>
            <ul>
              Price Range
              <li>
                <input
                  type="checkbox"
                  id={"price"}
                  value={20000}
                  onClick={handleFilter}
                />
                <label>0-20000</label>
              </li>
              <li>
                <input
                  type="checkbox"
                  id={"price"}
                  value={40000}
                  onClick={handleFilter}
                />
                <label>20000-40000</label>
              </li>
              <li>
                <input
                  type="checkbox"
                  id={"price"}
                  value={60000}
                  onClick={handleFilter}
                />
                <label>40000-60000</label>
              </li>
              <li>
                <input
                  type="checkbox"
                  id={"price"}
                  value={80000}
                  onClick={handleFilter}
                />
                <label>60000-80000</label>
              </li>
            </ul>
            <button onClick={mainFilter}className='btn btn-primary'>Apply filter</button>
          </div>
        </div>
      </CSSTransition>
      <Suspense fallback={<div>Loading</div>}>
        <ProductList list={products}></ProductList>
      </Suspense>
    </div>
  );
}
const mapStateToProps = (state) => {
  return {
    products: state.products,
    options: state.brands,
    productTypes: state.productTypes,
  };
};

export default connect(mapStateToProps)(About2);
