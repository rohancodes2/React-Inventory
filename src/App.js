import React,{lazy,Suspense} from "react";
import logo from "./logo.svg";
import "./App.css";
 import Navbar from "./components/Navbar";
import store from './Redux/Store'
import {Provider } from 'react-redux'
import { BrowserRouter as Router ,Route } from "react-router-dom";
import SignIn from './components/SignIn'
import Main from './components/Main'


function App() {
  return (
    <Provider store={store}>
   <div >
    
   <Navbar></Navbar>
    
 
  
   </div>
   </Provider>
  );
}

export default App;
