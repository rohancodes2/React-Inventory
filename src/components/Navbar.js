import React, { useEffect, useState } from "react";
import axios from 'axios'
import { connect } from 'react-redux';
import About from "./About";
import Signout from "./Signout";
import Products from "./Products";
import { Route, BrowserRouter as Router, NavLink, Link } from "react-router-dom";
import { MainItems,SubItems1,SubItems2,SignStatus } from "../data/NavbarItems";
import EditProfile from "./EditProfile";
import { initializeProduct } from "../Redux/Actions/ActionCreators";
import './Navbar.css';
import SignIn  from "./SignIn";
import { getToken, getUser, removeUserSession } from "../utils/mainUtils";
import Register from "./Register";
import uuid from "react-uuid";
import About2 from "./About2";
function Navbar(props) {
const [initialData,setData]=useState([]);
const [check,setcheck]=useState(false);
const [SubItems,setSubItems]=useState([]);
const [user,setUser]=useState('SignIn');
useEffect(()=>{
  console.log('hey useeffect nav')
  
  getToken()?setSubItems(SubItems1):setSubItems(SubItems2)
 if (getUser()!=null){setUser(getUser())};
  props.initializeProduct(initialData)
 
  axios.get('http://localhost:3000/products').then(function (response) {
       // handle success
   props.initializeProduct(response.data);
     
     })
     .catch(function (error) {
       // handle error
       console.log(error);
     })
     .then(function () {
       // always executed
     })}
    ,[])
  
  const handlerClick=(e)=>{
    //check if signed in 
   // SignContainer='w3-dropdown-hover w3-hide-small'
    //check if not signed in
    
    const check= (e.target.name==='About')?console.log("ok"): props.change;
    //const check2 = (e.target.value==='Sign In')?alert("Sign in"):SignContainer='w3-dropdown-hover w3-hide-small';
  }
  return (
     <Router>
    <div>
      {console.log("redux first list ",props.products)}
    <div className="w3-bar nnn w3-card" >
    
      <nav style={{ margin: 10 }}>
      {MainItems.map((item) => {
          return (
            <NavLink
            key={uuid()}
          to={item.url}
          className={item.class}
          name={item.name}
          onClick={handlerClick}
          activeClassName={'active'}
        >
          {item.name}
        </NavLink>

          );
        })}

  
      </nav>
      <div className='w3-right w3-dropdown-hover '>
  <button onClick={props.change}className=" btn-lg btn-outline-light text-white "
           title="More"> {user}
          <i className="fa fa-caret-down"></i>
          </button>
          <div className="w3-dropdown-content w3-bar-block w3-card-4">
            {
              SubItems.map((item) => {
              return (
                <NavLink
                key={uuid()}
                to={item.url}
                className={item.class}
              
                name={item.name}
                onClick={handlerClick}
              >
                {item.name}
              </NavLink>
                
              );
            })}
          </div>
        </div>
      </div>
    {console.log(getToken())}
    <Route path="/home"  exact component={About2} />
      <Route path="/products"  render={()=> getToken()?<Products/>:<SignIn/>} />
      <Route path="/signIn"component={SignIn}/>
      <Route path="/editprofile"  component={EditProfile} />
      <Route path="/register"  component={Register} />
      <Route path="/signout" render={()=><Signout/>} />
    </div>
    </Router> 
  );
          }

          
       

const mapStateToProps = state =>{
  return {
    products: state.products
  }
}
const mapDispatchToProps = dispatch =>{
  return {
    initializeProduct:prod=>dispatch(initializeProduct(prod))
  }
}


export default connect(mapStateToProps,mapDispatchToProps)(Navbar)