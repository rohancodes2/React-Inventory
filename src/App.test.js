import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Navbar from './components/Navbar';
import { mount, shallow } from './components/enzyme';

it('check if Navbar is rendered', () => {
  let wrapper = shallow(<App/>);
  
  expect(wrapper.find(Navbar).length).toEqual(1);
});
describe('full dom rendering', () => {
  let wrapper;

beforeEach(() => {

  wrapper= mount(<App/>);
})
it('full dom rendering', () => {

  expect(wrapper).toMatchSnapshot();
});


})