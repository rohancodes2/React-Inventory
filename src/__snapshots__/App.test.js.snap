// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full dom rendering full dom rendering 1`] = `ReactWrapper {}`;
